src/main/java/backend/resource/MultiModel.java

```diff
@@ -201,21 +201,34 @@ private void preprocessUpdatedIssues(List<Model> newModels) {
         for (Model model : newModels) {
             assert models.containsKey(model.getRepoId());
             Model existingModel = models.get(model.getRepoId());
-            if (!existingModel.getIssues().equals(model.getIssues())) {
-                // Find issues that have changed and update preferences with them
-                for (int i = 1; i <= model.getIssues().size(); i++) {
-                    // TODO O(n^2), optimise by preprocessing into a map or sorting
-                    if (!existingModel.getIssueById(i).equals(model.getIssueById(i))) {
-                        assert model.getIssueById(i).isPresent();
-                        // It's no longer currently read, but it retains its updated time.
-                        // No changes to preferences.
-                        model.getIssueById(i).get().setIsCurrentlyRead(false);
-                    }
-                }
+            if (existingModel.getIssues().equals(model.getIssues())) return;
+            for (int i = 1; i <= model.getIssues().size(); i++) {
+                if (existingModel.getIssueById(i).equals(model.getIssueById(i))) return;
+                // TODO O(n^2), optimise by preprocessing into a map or sorting
+                assert model.getIssueById(i).isPresent();
+                /** If the issue was marked read after the last update, but update came in later
+                 * due to network latency, ensure the issue retains its read state
+                **/
+                if (!wasNotJustMarkedAsRead(model.getIssueById(i).get())) return;
+                model.getIssueById(i).get().setIsCurrentlyRead(false);
             }
         }
     }
 
+    private boolean wasNotJustMarkedAsRead(TurboIssue issue) {
+        if (!issue.isCurrentlyRead() || !issue.getMarkedReadAt().isPresent()) return false;
+        LocalDateTime markedReadAt = issue.getMarkedReadAt().get();
+        LocalDateTime updatedAt = issue.getUpdatedAt();
+        if (updatedAt.isAfter(markedReadAt)) {
+            return true;
+        } else if (updatedAt.isBefore(markedReadAt) &&
+                   (markedReadAt.toEpochSecond(null) - updatedAt.toEpochSecond(null) < 3L)) {
```

Comment:
The zone offset [must be non-null](http://cr.openjdk.java.net/~sherman/jdk8_threeten/defaultM2/raw_files/new/src/share/classes/javax/time/chrono/ChronoLocalDateTime.java).

There is `Utility.localDateTimeToLong`, which I guess you could use. An alternative is the [`until` method](https://github.com/HubTurbo/HubTurbo/blob/master/src/main/java/filter/expression/Qualifier.java#L534-L535). That might be better actually.